### PR TITLE
Data action sample widget

### PR DESCRIPTION
**Pull request by xiaoyaohu0325**

### CODE REVIEW

File: samples/widgets/show-record-id/src/runtime/widget.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  LICENSE file.
+*/
+import {React, AllWidgetProps, IMState} from 'jimu-core';
+
+export interface Props {
+  featureIds?: string
+}
+
+export default class Widget extends React.PureComponent<AllWidgetProps & Props>{
+  static mapExtraStateProps = (
```
Comment (qlqllu) [inline samples/widgets/show-record-id/src/runtime/widget.tsx:27]: It seems we don't need the `mapExtraStateProps`, we can get it from props directly.

**Comment by qlqllu** [comment]
@dmartine can you help review as well?

**Comment by qlqllu** [comment]
@xiaoyaohu0325 The PR should be to the `1.4` branch.

File: samples/widgets/show-record-id/README.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# Show record id
+
+This sample demonstrates how to provide data action in a widget.
+
+## How to use the sample
+Clone the [sample repo](https://github.com/esri/arcgis-experience-builder-sdk-resources) and copy this widget's folder (within `samples/widgets`) to the `client/your-extensions/widgets` folder of your Experience Builder installation.
+
+## How it works
+1. Define your data action in the folder of `src/data-actions`. You should extend the base class `AbstractDataAction` and implement its methods: `isSupported` and `onExecute`.
```
Comment (gavinr) [inline samples/widgets/show-record-id/README.md:9]:
```suggestion
1. Define the data action in the folder of `src/data-actions`. Then extend the base class `AbstractDataAction` and implement its methods: `isSupported` and `onExecute`.
```

File: samples/widgets/show-record-id/README.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+2. Register your data action in manifest.
```
Comment (gavinr) [inline samples/widgets/show-record-id/README.md:10]:
```suggestion
2. Register the data action in manifest.
```

File: samples/widgets/show-record-id/README.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+```
+  "dataActions": [
+    {
+      "name": "showId",
+      "label": "show id",
+      "uri": "data-actions/show-id"
+    }
+  ],
+```
+3. Add nls labels in `src/runtime/translations/default.ts` for your data action if necessary. The name of the nls property should follow the pattern `_action_${dataActionName}_label`. In this case, it is `_action_showId_label`.
```
Comment (gavinr) [inline samples/widgets/show-record-id/README.md:20]:
```suggestion
3. Add nls labels in `src/runtime/translations/default.ts` for the data action if necessary. The name of the nls property should follow the pattern `_action_${dataActionName}_label`. In this case, it is `_action_showId_label`.
```

**Review by gavinr — APPROVED** [review]

**Review by dmartine — APPROVED** [review]
Looks good with suggestions from Gavin

**Review by qlqllu — APPROVED** [review]

**Pull request merged**